### PR TITLE
internal/{cli,server}: Introduce trigger token type

### PR DESCRIPTION
**Pull request by briancain**
This commit introduces a new token type for trigger URL actions and
includes a way to generate it via the user token CLI subcommand.
Currently, as of this pull request, authing with it does nothing and attempting to use it
against a gRPC endpoint will return an error permission denied.

### CODE REVIEW

File: internal/server/proto/server.proto [hunk 1]
```diff
@@ -3971,6 +3974,17 @@ message Token {
...
 
+  // The Trigger message is a kind of token type that is only used for
+  // authenticated trigger URL requests. It should not have any other
+  // authorized access to make requests in any other API endpoint.
+  message Trigger {
+    // The user that initiated the trigger token generation
+    string from_user_id = 1;
+
+    // The login token we'd like to create.
+    Login login = 2;
```
Comment (mitchellh) [inline internal/server/proto/server.proto:3985]: I don't think you need any fields in the `Trigger` message at all

**Review by mitchellh — APPROVED** [review]
One small change please but the rest is good

File: internal/server/proto/server.proto (hunk 1 above)
Comment (briancain) [reply internal/server/proto/server.proto:3985]: Ah yeah, I don't think we really need the login message. The user that created the token might be useful though?

Comment (mitchellh) [reply internal/server/proto/server.proto]: yeah that sounds good!

**Review by evanphx — APPROVED** [review]

**Comment by briancain** [comment]
Force pushed to fixup the merge conflicts with the server protos.

**Pull request merged**